packages/synapse-react-client/src/components/Authentication/LastLoginInfo.tsx

```diff
@@ -0,0 +1,194 @@
+import { InfoTwoTone } from '@mui/icons-material'
+import { Box, Link, Tooltip, Typography } from '@mui/material'
+import { useLocalStorageValue } from '@react-hookz/web'
+import dayjs from 'dayjs'
+import React from 'react'
+import {
+  CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+  LAST_LOGIN_DATE_LOCALSTORAGE_KEY,
+  LAST_LOGIN_METHOD_LOCALSTORAGE_KEY,
+  LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+  LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+} from '../../utils/SynapseConstants'
+import { formatDate } from '../../utils/functions/DateFormatter'
+import { LoginMethod, getLoginMethodFriendlyName } from './LoginMethod'
+
+export type LastLoginInfoProps = {
+  currentSourceAppName: string | undefined
+  lastLoginMethod: LoginMethod | undefined
+  lastLoginSourceAppName: string | undefined
+  lastLoginSourceAppURL: string | undefined
+  lastLoginDate: string | undefined
+  display: 'sentence' | 'box'
+}
+
+export function useLastLoginInfoStatePairs() {
+  return {
+    currentSourceAppNameStatePair: useLocalStorageValue<string>(
+      CURRENT_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    ),
+    lastLoginMethodStatePair: useLocalStorageValue<LoginMethod>(
+      LAST_LOGIN_METHOD_LOCALSTORAGE_KEY,
+    ),
+    lastLoginSourceAppNameStatePair: useLocalStorageValue<string>(
+      LAST_LOGIN_SOURCEAPP_NAME_LOCALSTORAGE_KEY,
+    ),
+    lastLoginSourceAppURLStatePair: useLocalStorageValue<string>(
+      LAST_LOGIN_SOURCEAPP_URL_LOCALSTORAGE_KEY,
+    ),
+    lastLoginDateStatePair: useLocalStorageValue<string>(
+      LAST_LOGIN_DATE_LOCALSTORAGE_KEY,
+    ),
+  }
+}
```

Comment:
Just being picky at this point but why `pairs` in the name? I think you would be fine with `useLastLoginInfoState` and e.g. `currentSourceAppNameState`